src/react/utils.js

```diff
@@ -591,12 +604,36 @@ export function setProperty(
   let binding;
   if (typeof property === "string") {
     binding = object.properties.get(property);
-  } else {
+    if (!binding) {
+      binding = {
+        descriptor: {
+          configureable: true,
+          enumerable: true,
+          writable: true,
+          value,
+        },
+        key: property,
+        object,
+      };
+      object.properties.set(property, binding);
+    }
+  } else if (property instanceof SymbolValue) {
     binding = object.symbols.get(property);
+    if (!binding) {
+      binding = {
```

Comment:
You seem to create the binding the same way in both branches. Maybe unify that part?